medmodels/treatment_effect/matching/tests/test_matching.py

```diff
@@ -0,0 +1,241 @@
+            medrecord=self.medrecord,
+            control_set=control_set,
+            treated_set=treated_set,
+            patients_group="patients",
+            essential_covariates=["age", "gender"],
+            one_hot_covariates=["gender"],
+        )
+
+        # Assert that the treated and control dataframes have the correct columns
+        self.assertIn("age", data_treated.columns)
+        self.assertIn("age", data_control.columns)
+        self.assertTrue(
+            "gender_female" in data_treated.columns
+            or "gender_male" in data_treated.columns
+        )
+        self.assertTrue(
+            "gender_female" in data_control.columns
+            or "gender_male" in data_control.columns
+        )
+
+        # Assert that the treated and control dataframes have the correct number of rows
+        self.assertEqual(len(data_treated), len(treated_set))
+        self.assertEqual(len(data_control), len(control_set))
+
+        # Try automatic detection of attributes
+        data_treated, data_control = neighbors_matching._preprocess_data(
+            medrecord=self.medrecord,
+            control_set=control_set,
+            treated_set=treated_set,
+            patients_group="patients",
+        )
+
+        # Assert that the treated and control dataframes have the correct columns
+        self.assertIn("age", data_treated.columns)
+        self.assertIn("age", data_control.columns)
+        self.assertTrue(
+            "gender_female" in data_treated.columns
+            or "gender_male" in data_treated.columns
+        )
+        self.assertTrue(
+            "gender_female" in data_control.columns
+            or "gender_male" in data_control.columns
+        )
+
+        # Assert that the treated and control dataframes have the correct number of rows
+        self.assertEqual(len(data_treated), len(treated_set))
+        self.assertEqual(len(data_control), len(control_set))
+
+    def test_match_controls(self):
+        neighbors_matching = NeighborsMatching(number_of_neighbors=1)
+
+        control_set: Set[NodeIndex] = {"P1", "P3", "P5", "P7", "P9"}
+        treated_set: Set[NodeIndex] = {"P2", "P4", "P6"}
+
+        matched_controls = neighbors_matching.match_controls(
+            medrecord=self.medrecord,
+            control_set=control_set,
+            treated_set=treated_set,
+            patients_group="patients",
+            essential_covariates=["age", "gender"],
+            one_hot_covariates=["gender"],
+        )
+
+        # Assert that the matched controls are a subset of the control set
+        self.assertTrue(matched_controls.issubset(control_set))
+
+        # Assert that the correct number of controls were matched
+        self.assertEqual(len(matched_controls), len(treated_set))
+
+        # Assert it works equally if no covariates are given (automatically assigned)
+        matched_controls_no_covariates_specified = neighbors_matching.match_controls(
```

Comment:
Maybe add checking if `matched_controls_no_covariates_specified` and `matched_controls ` are equal?